controller/api/models.py

```diff
@@ -292,6 +292,7 @@ def _start_containers(self, to_add):
         if any(c.state != 'created' for c in to_add):
             err = 'aborting, failed to create some containers'
             log_event(self, err, logging.ERROR)
+            self._destroy_containers(to_add)
```

Comment:
Oh wait... That's not how exceptions work. The old containers won't be destroyed because an exception was raised, causing the rest of the block to not be interpreted until the RuntimeError is dealt with. Still, I do believe that we should be handling this better in .scale() and in .deploy() rather than in ._start_containers().